libvast/src/system/disk_monitor.cpp

```diff
@@ -142,6 +142,10 @@ disk_monitor(disk_monitor_actor::stateful_pointer<disk_monitor_state> self,
                         erased_ids)
               .then(
                 [=, sg = shared_guard](atom::done) {
+                  // TODO: There's a race condition here: We calculate the size
+                  // of the database directory while we might be deleting files
+                  // from it. Both the size calculation and file delete
+                  // operations must go through the filesystem actor instead.
```

Comment:
Let's keep the solution space more open until we have agreed on how to solve the issue internally.

Reply:
Will reword.